jobs/ServiceInstanceUpdateJob.js

```diff
@@ -134,41 +146,49 @@ class ServiceInstanceUpdateJob extends BaseJob {
       });
   }
 
-  static getOutdatedDiff(deploymentName, instanceId, plan) {
+  static deploymentStaggered(err) {
+    const response = _.get(err, 'error', {});
+    const description = _.get(response, 'description', '');
+    return description.indexOf(CONST.FABRIK_OPERATION_STAGGERED) > 0 && description.indexOf(CONST.FABRIK_OPERATION_COUNT_EXCEEDED) > 0;
+  }
+
+  static deploymentLocked(err) {
+    const response = _.get(err, 'error', {});
+    const description = _.get(response, 'description', '');
+    return description.indexOf(CONST.OPERATION_TYPE.LOCK) > 0;
+  }
+
+  static getOutdatedDiff(deploymentName, tenantInfo, plan) {
     return Fabrik
       .createManager(plan)
-      .then((directorManager) => cloudController.getOrgAndSpaceGuid(instanceId)
-        .then(opts => {
-          const context = {
-            platform: CONST.PLATFORM.CF,
-            organization_guid: opts.organization_guid,
-            space_guid: opts.space_guid
-          };
-          opts.context = context;
-          return directorManager
-            .diffManifest(deploymentName, opts);
-        })
-      );
+      .then(directorManager => directorManager.diffManifest(deploymentName, tenantInfo));
   }
 
-  static updateDeployment(deploymentName, diff, skipForbiddenCheck, runImmediately) {
+  static updateServiceInstance(instance_id, diff, skipForbiddenCheck, runImmediately, plan, context) {
     return Promise
       .try(() => {
         if (!skipForbiddenCheck) {
           utils.hasChangesInForbiddenSections(diff);
         }
       })
-      .then(() => Fabrik
-        .createOperation('update', {
-          deployment: deploymentName,
-          username: 'Auto_Update_Job',
-          arguments: {},
-          runImmediately: runImmediately || false
+      .then(() => serviceBrokerClient
+        .updateServiceInstance({
+          instance_id: instance_id,
+          context: context,
+          service_id: plan.service.id,
+          plan_id: plan.id,
+          previous_values: {
+            service_id: plan.service.id,
+            plan_id: plan.id,
+            organization_id: context.organization_id,
+            space_id: context.space_guid
+          },
+          parameters: {
+            scheduled: true,
```

Comment:
We should pass and preserve the `service-fabrik-operation` parameter. `EventlogInterceptor` has dependency on the same.